common/models/user.js

```diff
@@ -669,6 +669,42 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function beforeEmailUpdate(ctx, next) {
+      var oldEmail, newEmail, oldId, newId;
+      if (ctx.isNewInstance) return next();
+
+      if (ctx.where) {
+        ctx.hookState.id = ctx.where.id;
+        oldId = ctx.hookState.id;
+      }
+
+      ctx.Model.find({ where: { id: oldId }}, function(err, userInstances) {
+        ctx.hookState.originalUserData = userInstances.map(function(u) {
+          return { id: u.id, email: u.email };
+        });
+      });
+      next();
+
+      UserModel.observe('after save', function afterEmailUpdate(ctx, next) {
```

Comment:
Explicit ordering here sounds like a potential source of confusing bugs. I'm imagining a scenario where the tokens are deleted in one hook and then a later hook tries to validate one of the deleted tokens and fails, resulting in the remainder of the operation failing and leaving things in a state where here is no valid token left to retry with.

I'm not intimately familiar with loopback, so the above is from a generic security token handling perspective.